engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityCache.java

```diff
@@ -0,0 +1,350 @@
+        if (entityManager.isExistingEntity(id)) {
+            return createEntityRef(id);
+        }
+        return EntityRef.NULL;
+    }
+
+    /**
+     * Creates the entity without sending any events. The entity life cycle subscriber will however be informed.
+     */
+    @Override
+    public EntityRef createEntityWithoutLifecycleEvents(Iterable<Component> components) {
+        EntityRef entity = createEntity(components);
+        for (Component component: components) {
+            entityManager.notifyComponentAdded(entity, component.getClass());
+        }
+        return entity;
+    }
+
+    /**
+     * Creates the entity without sending any events. The entity life cycle subscriber will however be informed.
+     */
+    @Override
+    public EntityRef createEntityWithoutLifecycleEvents(String prefabName) {
+        return createEntityWithoutLifecycleEvents(entityManager.getPrefabManager().getPrefab(prefabName));
+    }
+
+    /**
+     * Creates the entity without sending any events. The entity life cycle subscriber will however be informed.
+     */
+    @Override
+    public EntityRef createEntityWithoutLifecycleEvents(Prefab prefab) {
+        if (prefab != null) {
+            List<Component> components = Lists.newArrayList();
+            for (Component component : prefab.iterateComponents()) {
+                components.add(entityManager.getComponentLibrary().copy(component));
+            }
+            components.add(new EntityInfoComponent(prefab, prefab.isPersisted(), prefab.isAlwaysRelevant()));
+
+            return createEntityWithoutLifecycleEvents(components);
+        } else {
+            return createEntityWithoutLifecycleEvents(Collections.<Component>emptyList());
+        }
+    }
+
+    /**
+     * Destroys the entity without sending any events. The entity life cycle subscriber will however be informed.
+     */
+    @Override
+    public void destroyEntityWithoutEvents(EntityRef entity) {
+        if (entity.isActive()) {
+            entityManager.notifyComponentRemovalAndEntityDestruction(entity.getId(), entity);
+            destroy(entity);
+        }
+    }
+
+    @Override
+    public EntityRef createEntityWithId(long id, Iterable<Component> components) {
+        if (!entityManager.registerId(id)) {
+            return EntityRef.NULL;
+        }
+        for (Component c : components) {
+            componentStore.put(id, c);
+        }
+        EntityRef entity = createEntityRef(id);
+        if (entityManager.getEventSystem() != null) {
+            entityManager.getEventSystem().send(entity, OnActivatedComponent.newInstance());
+        }
+        for (Component component: components) {
+            entityManager.notifyComponentAdded(entity, component.getClass());
+        }
+        return entity;
+    }
+
+    public EntityBuilder newBuilder() {
+        return new EntityBuilder(this);
+    }
+
+    @Override
+    public EntityBuilder newBuilder(String prefabName) {
```

Comment:
The builder class says that the whole entity can be setup before creation without any events being sent in-between. It also seems like a good way of creating lots of the same entity, so might be useful for spawning lots of cities, for example.

Implementing some of the create() methods in terms of the builder should reduce code duplication, especially the code to extract components from a prefab.